### DISCUSSION TITLE
Solved: Cmd + click not working?

### DISCUSSION THREAD

**Discussion by luckman212** [Taitava/obsidian-shellcommands#346] [Debugging and testing]
I just started playing with this amazing plugin. 

I wanted to try Cmd+click to redirect the output to one of my wrappers, but it doesn't seem to have any effect. The command still runs in its default context.

- Obsidian 1.2.8 (installer 1.2.7)
- macOS 13.3.1

<img width="748" alt="screenshot_AwRNevjM_gh" src="https://user-images.githubusercontent.com/1992842/236062877-aad0b5ca-c1c5-445b-b2e5-c4dd78532c9b.png">

**Comment by Taitava** [comment]
Hi! 🙂 Thank you for your report. I don't have a Mac, so I was not able to test `Cmd` clicking when I developed the _Execute now and ask what to do with output_ feature. So yes, this is a bug and I will fix it. Meanwhile, you can try to press Mac's `Ctrl` while clicking. If that doesn't work, please try also `Option`. I can't remember right now how I programmed this feature.

I'll create a bug issue for this. I'll probably borrow a Mac from a friend of mine so I can better fix this.

Btw. The _ask what to do with output_ feature doesn't change [output wrappers](https://publish.obsidian.md/shellcommands/Output+handling/Output+wrappers). It just [shows all output in a modal](https://publish.obsidian.md/shellcommands/Output+handling/Output+channel+-+Ask+after+execution) and you can then select where to use the output. In case you have defined an output wrapper for the shell command in settings, the wrapper text will show up in the modal, too, but there's no way to change the wrapper in the modal.

**Comment by luckman212** [comment]
Thanks @Taitava - yes I tried other modifier keys as well (Ctrl, Option, Shift) but they also had no effect.

It would be cool to allow the wrapper to be selected on the fly, but I understand that's probably an edge case.

**Reply by Taitava** [reply]
Ok, thanks for the information about the other keys. I can't say when I can get to fix this, currently my time is unfortunately quite limited.

I'll keep the wrpper selection idea on my mind and think about if it's doable, or if it's too edge case. Personally, I haven't used output wrappers much myself, so I haven't thought much about all the possible use cases for them. So it's good that someone brings up ideas I'd never think about myself. 🙂👍

**Comment by Taitava** [comment]
# Test fix 

I'm testing to fix the bug. @luckman212 If you want, you can test it too, before I'll make an actual release of the fix (version `0.20.1`). Just download the zip file below and replace main.js in `.obsidian/plugins/obsidian-shellcommands`. styles.css was not changed, although it's in the zip, too.
[SC-cmd-click-fix.zip](https://github.com/Taitava/obsidian-shellcommands/files/12329844/SC-cmd-click-fix.zip)

**Comment by luckman212** [comment]
@Taitava I just got the chance to test this. It works! 🎉 Thanks for the fix.

Only thing I am seeing are 2 errors in the console log:

<img width="894" alt="screenshot_CiPerCAo" src="https://github.com/Taitava/obsidian-shellcommands/assets/1992842/c2879bd9-fa87-4c04-a99d-a01895e678ed">

This is with 1.4.2 (Installer 1.3.7)

**Reply by Taitava** [reply]
Thanks! 🌞

And [the errors seem to be the same ones you encountered earlier](https://github.com/Taitava/obsidian-shellcommands/discussions/348), so this fix seems to work fine regardless of them 🙂. I haven't had time to look at that those errors yet, so this fix did not even try to fix them. 🙂

**Reply by luckman212** [reply]
Ah yes, that's true. Forgot I'd even reported those 😐

**Comment by Taitava** [comment]
# Released
The click bug fix is now finally released in version `0.20.1`. No need to use the test fix zip file anymore. 🙂
